src/wp-admin/authorize-application.php

```diff
@@ -11,29 +11,21 @@
 
 $error        = null;
 $new_password = '';
+$user         = wp_get_current_user();
 
 // This is the no-js fallback script. Generally this will all be handled by `auth-app.js`.
 if ( isset( $_POST['action'] ) && 'authorize_application_password' === $_POST['action'] ) {
 	check_admin_referer( 'authorize_application_password' );
 
-	$success_url = $_POST['success_url'];
-	$reject_url  = $_POST['reject_url'];
-	$app_name    = $_POST['app_name'];
-	$app_id      = $_POST['app_id'];
-	$redirect    = '';
-
+	$redirect = '';
 	if ( isset( $_POST['reject'] ) ) {
-		if ( $reject_url ) {
-			$redirect = $reject_url;
-		} else {
-			$redirect = admin_url();
-		}
+		$redirect = $_POST['reject_url'] ?? admin_url();
```

Comment:
This really should be run through `wp_unslash()`:


```suggestion
		$redirect = isset( $_POST['reject_url'] ) ? wp_unslash( $_POST['reject_url'] ) : admin_url();
```